src/snippet/mod.rs

```diff
@@ -186,6 +186,29 @@ fn select_best_fragment_combination(fragments: &[FragmentCandidate], text: &str)
     }
 }
 
+/// Returns ranges that are collapsed into non-overlapped ranges.
+fn collapse_overlapped_ranges(ranges: &Vec<Range<usize>>) -> Vec<Range<usize>> {
+    let mut result = Vec::new();
+    let mut ranges = ranges.iter();
+
+    let mut current = match ranges.next() {
+        Some(range) => range.clone(),
+        None => return result,
+    };
+
+    for range in ranges {
+        if current.end > range.start {
+            current = current.start..range.end;
```

Comment:
nitpick... I feel the following is easier but that's your call.
```suggestion
            current.end = range.end;
```